docs/source-2.0/aws/amazon-eventstream.rst

```diff
@@ -0,0 +1,430 @@
+
+Additionally, the message has the following size restrictions:
+
+* The payload of a message MUST NOT exceed 25,165,824 bytes (24 MB).
+* The encoded headers of a message MUST NOT exceed 131,072 bytes (128 kB). Note
+  that the in-memory representation of the headers, once decoded, may legally
+  exceed this value.
+
+Services MUST validate these additional size restrictions. Clients MUST NOT
+validate them.
+
+.. _amazon-eventstream-headers-format:
+
+Headers Format
+==============
+
+Headers are used to annotate a message with arbitrary metadata, such as
+timestamps, message type information, additional payload checksums, and so on.
+Headers are key-value pairs where the key is a UTF-8 string and the value is one
+of several possible types. Note that this is distinct from HTTP headers, whose
+values are always bytes (usually UTF-8 encoded strings). Any given header name
+MUST only appear once in a message. The oder in which headers are encoded is
+meaningless; implementations MAY NOT preserve header order across encoding or
+decoding.
+
+Header values are prefixed with a type. The set of types is fixed and not open
+to extension. Implementations SHOULD bind header types to the corresponding
+types in their programming language.
+
+.. code-block:: text
+
+    struct {
+        struct {
+            utf8 name<1..255>;
+        } header_name;
+        HeaderValueType type;
+        HeaderValue header_value;
+    } Header;
+
+    enum {
+        boolean_true(0),
+        boolean_false(1),
+        byte(2),
+        short(3),
+        integer(4),
+        long(5),
+        byte_array(6),
+        string(7),
+        timestamp(8),
+        uuid(9)
+    } HeaderValueType;
+
+    struct {
+        select (HeaderValueType) {
+            case boolean_true:
+            case boolean_false:
+                struct {};
+            case byte:
+                int8 value;
+            case short:
+                int16 value;
+            case integer:
+                int32 value;
+            case long:
+                int64 value;
+            case byte_array:
+                byte data<1..2^15-1>;
+            case string:
+                utf8 data<1..2^15-1>;
+            case timestamp:
+                int64 millis_since_epoch;
+            case uuid:
+                byte value[16];
+        } value;
+    } HeaderValue;
+
+.. _amazon-eventstream-semantics:
+
+----------------------
+Amazon Event Semantics
+----------------------
+
+The ``application/vnd.amazon.eventstream`` wire format is a generic encoding.
+There are additional semantic requirements when using this format with AWS
+protocols. Events are categorized into one of five categories: message events,
+modeled error events, unmodeled error events, initial request events, and
+initial response events. Each event category has its own semantic requirements.
+
+Message Events
+==============
+
+Message events are events that represent an event union member whose target does
```

Comment:
This is the first mention of a `union` for event streams in this doc. It's probably worth either an introduction, or at least a link to the event streaming model spec.